File: app/src/main/java/com/example/easywheel/algorithm/AccessibilityRouteScorer.java
package com.example.easywheel.algorithm;

public class AccessibilityRouteScorer {

    // Score route based on accessibility features
    public static int calculateScore(boolean hasRamp,
                                     boolean smoothPath,
                                     boolean obstacle,
                                     boolean stairs) {

        int score = 50; // base score

        if (hasRamp) {
            score -= 10;
        }

        if (smoothPath) {
            score -= 5;
        }

        if (obstacle) {
            score += 20;
        }

        if (stairs) {
            score += 100;
        }

        return score;
    }
}

File: app/src/main/java/com/example/easywheel/AccessibilityRouteScorer.kt
package com.example.easywheel

object AccessibilityRouteScorer {

    fun calculateScore(
        hasRamp: Boolean,
        smoothPath: Boolean,
        obstacle: Boolean,
        stairs: Boolean
    ): Int {

        var score = 100

        if (!hasRamp) score -= 30
        if (!smoothPath) score -= 20
        if (obstacle) score -= 25
        if (stairs) score -= 40

        return score
    }
}